pkg/services/govmomi/vcenter/clone.go

```diff
@@ -390,6 +404,105 @@ func getDiskConfigSpec(disk *types.VirtualDisk, diskCloneCapacityKB int64) (type
 	}, nil
 }
 
+// createDataDisks parses through the list of VSphereDisk objects and generates the VirtualDeviceConfigSpec for each disk needing to be created during clone process.
+func createDataDisks(ctx context.Context, disks []infrav1.VSphereDisk, primaryDisk *types.VirtualDisk, devices object.VirtualDeviceList) ([]types.BaseVirtualDeviceConfigSpec, error) {
+	log := ctrl.LoggerFrom(ctx)
+	additionalDisks := []types.BaseVirtualDeviceConfigSpec{}
+	unit := int32(1)
+
+	for i, dataDisk := range disks {
+		log.Info("Adding disk", "spec", dataDisk)
```

Comment:
this is an example of what i meant from earlier. seems having some name / id will definitely help especially during debugging.

Reply:
Would it make sense to add `deviceName` or `deviceId` as a field for each disk entry?

Reply:
if we can define an optional string name on each disk in CAPV and that can propagate in the govmomi API types that would make the UX a little bit nicer i would argue.
if its not possible i think it's worth printing the index of the disk as defined in the slice of disks ini the CAPV types in all logged entries in controllers.

e.g. `doing X with disk at index Y`

Reply:
Sounds good.  I'll go ahead and add the index output in the log message and will update this later if we find a way to leverage a name.